datafusion/sqllogictest/src/engines/datafusion_engine/normalize.rs

```diff
@@ -96,14 +96,18 @@ fn expand_row(mut row: Vec<String>) -> impl Iterator<Item = Vec<String>> {
         // form new rows with each additional line
         let new_lines: Vec<_> = lines
             .into_iter()
-            .map(|l| {
+            .enumerate()
+            .map(|(idx, l)| {
                 // replace any leading spaces with '-' as
                 // `sqllogictest` ignores whitespace differences
                 //
                 // See https://github.com/apache/arrow-datafusion/issues/6328
                 let content = l.trim_start();
```

Comment:
I wonder if we can remove the `----` stuff now

Maybe we could remove the `---` and see if the test diff look better 🤔  

Not in this PR